docs/source/cc_service.md

```diff
@@ -0,0 +1,297 @@
+# Chaincode as an external service
+
+Fabric v2.0 supports chaincode deployment and execution outside of Fabric that enables users to manage a chaincode runtime independently of the peer. This facilitates deployment of chaincode on Fabric cloud deployments such as Kubernetes. Instead of installing the chaincode on every peer, chaincode can now run as a service as opposed to a Fabric launched client. This capability leverages the Fabric v2.0 external builder and launcher functionality which enables operators to extend a peer with programs to build, launch, and discover chaincode. Before reading this topic you should become familiar with the [External Builds and Launchers](./cc_launcher.html) content.
+
+Prior to the availability of the external build and launcher functionality, the chaincode package would contain the source code for the chaincode. The new external build and launcher functionality now allows users to optionally customize the build process. And with respect to running the chaincode as an external service, the build process allows you to specify the endpoint information of the server where the chaincode is running. Hence the package simply consists of the externally running chaincode server endpoint information and TLS artifacts for secure connection. TLS is optional but highly recommended for all environments except a simple test environment.
+
+The rest of this topic describes how to configure chaincode as an external service:
+
+* [Packaging chaincode](#packaging-chaincode)
+* [Configuring a peer to process external chaincode](#configuring-a-peer-to-process-external-chaincode)
+* [External builder and launcher sample scripts](#external-builder-and-launcher-sample-scripts)
+* [Writing chaincode to run as an external service](#writing-chaincode-to-run-as-an-external-service)
+* [Deploying the chaincode](#deploying-the-chaincode)
+* [Running the chaincode as an external service](#running-the-chaincode-as-an-external-service)
+
+## Packaging chaincode
+
+With the Fabric v2.0 chaincode lifecycle, chaincode is packaged and installed in a `.targz` format. The following `myccpackage.tgz` archive  demonstrates the required structure:
+
+```sh
+$ tar xvfz myccpackage.tgz
+code.tar.gz
+metadata.json
+```
+
+### code.tar.gz archive requirements
+
+The `code.tar.gz` archive must include a `connection.json` file which contains the chaincode endpoint and certificate information:
```

Comment:
This comes off as a little overly prescriptive to me.  Including a `connection.json` is the most obvious way to do it, but you could encode this information however you want, the information could come from a network resource, it really just matters that your external builder can produce a `connection.json` during its release step.

Overall, I think this section could be better restructured as 'an example external builder for a

Reply:
Agreed. How about something like this 
```
The `code.tar.gz` archive must include information obtaining connection information for the chaincode endpoint. This information is packaged into `connection.json` in the `release` step (see below). In this example we package `connection.json` directly in `code.tar.gz` so the `release` step can just copy it over.
```

Reply:
Agree with Jason here. There's a lot of flexibility, adding `connection.json` as a code artifact is _one_ way; you could also include an empty `code.tar.gz` and put the connection information into `metadata.json` to extract as necessary in build/release.

Reply:
I changed it to
`The `code.tar.gz` archive must include connection information for the chaincode endpoint. This information is packaged into `connection.json` in the `release` step (see below). In this example we package `connection.json` directly in `code.tar.gz` so the `release` step can just copy it over.`